webpack.config.js

```diff
@@ -1,77 +1,51 @@
 const path = require('path');
-const webpack = require('webpack');
-
-
-/**
- * Define plugins based on environment
- * @param {boolean} isDev If in development mode
- * @return {Array}
- */
-function getPlugins(isDev) {
-
-  const plugins = [
-    new webpack.optimize.OccurenceOrderPlugin(),
-    new webpack.DefinePlugin({}),
-  ];
-
-  if (isDev) {
-    plugins.push(new webpack.NoErrorsPlugin());
-  } else {
-    plugins.push(new webpack.optimize.DedupePlugin());
-    plugins.push(new webpack.optimize.UglifyJsPlugin({
-      minimize: true,
-      sourceMap: false,
-      compress: {
-        warnings: false,
-      },
-    }));
-  }
-
-  return plugins;
-
-}
-
 
 /**
  * Define loaders
  * @return {Array}
  */
-function getLoaders() {
-
-  const loaders = [{
-    test: /(\.js)/,
-    exclude: /(node_modules)/,
-    loaders: ['babel'],
-  }, {
-    test: /(\.jpg|\.png)$/,
-    loader: 'url-loader?limit=10000',
-  }, {
-    test: /\.json/,
-    loader: 'json-loader',
-  }];
-
-  return loaders;
-
+function getRules() {
+  return [
+    {
+      test: /(\.js)/,
+      exclude: /(node_modules)/,
+      use: {
+        loader: 'babel-loader',
+      },
+    },
+    {
+      test: /(\.jpg|\.png)$/,
+      use: [
+        {
+          loader: 'url-loader',
+          options: {
+            limit: 10000,
+          },
+        },
+      ],
+    },
+    {
+      test: /\.json/,
+      loader: 'json-loader',
+    },
+  ];
 }
 
-
-module.exports = (config) => {
+module.exports = config => {
```

Comment:
since node widely supports arg destructuring you could rewrite this and others:

`module.exports = ({ dev }) => { ... }`